backend/python/tapa/floorplan.py

```diff
@@ -0,0 +1,248 @@
+    )
+
+  _logger.info('generating partitioning constraints')
+  v2s, s2v, topology = tapa.autobridge.generate_floorplan(
+      top_task,
+      work_dir,
+      fifo_width_getter,
+      directive['connectivity'],
+      directive['part_num'],
+      **kwargs,
+  )
+
+  # extract the mapping from vertice to region
+  # inst_to_region = _get_instance_to_region(floorplan)
+  check_floorplan_validity(top_task, ctrl_instance_name, v2s, s2v)
+
+  # extract the pblock definition tcl for each region
+  region_to_pblock_tcl = _get_pblock_tcl(topology)
+
+  # determine partitioning of each FIFO, assign each pipeline register to a region
+  fifo_pipeline_level, fifo_pipeline_reg_to_region = get_fifo_pipelines(
+    top_task, v2s, topology)
+  inst_to_region = {**v2s, **fifo_pipeline_reg_to_region}
+
+  return inst_to_region, fifo_pipeline_level, region_to_pblock_tcl
+
+
+def get_post_synth_area(
+    work_dir,
+    cpp_dir,
+    rtl_dir,
+    directive,
+    top_task,
+    task_getter: Callable[[str], Task],
+):
+  def worker(module_name: str, idx: int) -> report.HierarchicalUtilization:
+    _logger.debug('synthesizing %s', module_name)
+    rpt_path = f'{work_dir}/report/{module_name}.hier.util.rpt'
+
+    rpt_path_mtime = 0.
+    if os.path.isfile(rpt_path):
+      rpt_path_mtime = os.path.getmtime(rpt_path)
+
+    # generate report if and only if C++ source is newer than report.
+    if os.path.getmtime(get_cpp(cpp_dir, module_name)) > rpt_path_mtime:
+      os.nice(idx % 19)
+      with report.ReportDirUtil(
+          rtl_dir,
+          rpt_path,
+          module_name,
+          directive['part_num'],
+          synth_kwargs={'mode': 'out_of_context'},
+      ) as proc:
+        stdout, stderr = proc.communicate()
+
+      # err if output report does not exist or is not newer than previous
+      if (not os.path.isfile(rpt_path) or
+          os.path.getmtime(rpt_path) <= rpt_path_mtime):
+        sys.stdout.write(stdout.decode('utf-8'))
+        sys.stderr.write(stderr.decode('utf-8'))
+        raise InputError(f'failed to generate report for {module_name}')
+
+    with open(rpt_path) as rpt_file:
+      return report.parse_hierarchical_utilization_report(rpt_file)
+
+  _logger.info('generating post-synthesis resource utilization reports')
+  with futures.ThreadPoolExecutor(max_workers=os.cpu_count()) as executor:
+    for utilization in executor.map(
+        worker,
+        {x.task.name for x in top_task.instances},
+        itertools.count(0),
+    ):
+      # override self_area populated from HLS report
+      bram = int(utilization['RAMB36']) * 2 + int(utilization['RAMB18'])
+      task_getter(utilization.instance).total_area = {
+          'BRAM_18K': bram,
+          'DSP': int(utilization['DSP Blocks']),
+          'FF': int(utilization['FFs']),
+          'LUT': int(utilization['Total LUTs']),
+          'URAM': int(utilization['URAM']),
+      }
+
+
+def _get_pblock_tcl(topoogy):
+  tcl = []
+  for props in topoogy.values():
```

Comment:
What is `props`? We prefer to [use descriptive names and avoid abbreviations](https://google.github.io/styleguide/pyguide.html#316-naming). Type annotations are helpful, too.

Also, there is a typo in `topology`.